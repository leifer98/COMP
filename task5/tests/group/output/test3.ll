
declare i32 @scanf(i8*, ...)
declare i32 @printf(i8*, ...)
declare void @exit(i32)
@.int_specifier_scan = constant [3 x i8] c"%d\00"
@.int_specifier = constant [4 x i8] c"%d\0A\00"
@.str_specifier = constant [4 x i8] c"%s\0A\00"
define i32 @readi(i32) {
%ret_val = alloca i32
%spec_ptr = getelementptr [3 x i8], [3 x i8]* @.int_specifier_scan, i32 0, i32 0
call i32 (i8*, ...) @scanf(i8* %spec_ptr, i32* %ret_val)
%val = load i32, i32* %ret_val
ret i32 %val
}
define void @printi(i32) {
%spec_ptr = getelementptr [4 x i8], [4 x i8]* @.int_specifier, i32 0, i32 0
call i32 (i8*, ...) @printf(i8* %spec_ptr, i32 %0)
ret void
}
define void @print(i8*) {
%spec_ptr = getelementptr [4 x i8], [4 x i8]* @.str_specifier, i32 0, i32 0
call i32 (i8*, ...) @printf(i8* %spec_ptr, i8* %0)
ret void
}
define i8 @foo() {
%t0 = add i8 0, 4
ret i8 %t0
}
define void @main() {
%t1 = add i32 0, 0
%pippo = alloca i32 
store i32 %t1, i32* %pippo
%t2 = call i8 @foo()
%t3 = zext i8 %t2 to i32
%foonum = alloca i32 
store i32 %t2, i32* %foonum
Visiting While Node
ret void
}
