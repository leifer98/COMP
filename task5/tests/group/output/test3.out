lli: <stdin>:36:11: error: '%t2' defined with type 'i8' but expected 'i32'
store i32 %t2, i32* %foonum
          ^
